packages/react/src/utils/anonymous.ts

```diff
@@ -0,0 +1,5 @@
+import { BaseUser, AnonymousUser } from "@slashid/slashid";
+
+export const userIsAnonymous = (user: BaseUser): user is AnonymousUser => {
```

Comment:
Wondering if this is needed, what is the added value on top of working with the `user` directly and checking `user.anonymous`? It's a documented public property

Reply:
It’s for type narrowing, so we don’t need to cast.

`User | AnonymousUser | undefined` -> `AnonymousUser`

Anon users have a unique method: `id()`. Calling this safely was the primary goal, anywhere else I used it was just for consistency.

Reply:
@jake-slashid is this correct though? A `BaseUser` that is anonymous could have `user.anonymous` set to `true`, but not be an instance of `AnonymousUser` 🤔

Reply:
Yeah good point!